command/agent/node_endpoint.go

```diff
@@ -205,7 +205,7 @@ func (s *HTTPServer) nodePurge(resp http.ResponseWriter, req *http.Request, node
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
 	args := structs.NodeDeregisterRequest{
-		NodeID: nodeID,
+		NodeIDs: []string{nodeID},
```

Comment:
We need to worry about backward compatibility and server compatibility here as well; consider the case where a new nomad client running against old server cluster.

Reply:
I think that in either the case of changing the message (because it was a change by addition) or the case that's in the pr now, of introducing a new message, this works fine.

Reply:
I think we still need to address this - as the http agent runs on every node (servers or clients) and needs to support communicating with other servers that are older.